PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -83,8 +84,26 @@ function transformReports(inputReportFiles: string[], index: number)
         sarifReports.push(report);
         processResults(inputReportFiles, index);
     } else if (report.toLocaleLowerCase().endsWith(XML_EXTENSION)) {
+        ruleAnalyzerPairs.clear();
+
         const saxStream = sax.createStream(true, {});
         saxStream.on("opentag", function (node) {
+            if (node.name == 'Rule' && !bLegacyReport) {
+                ruleAnalyzerPairs.set(node.attributes.id, node.attributes.analyzer);
+            }
+
+            if (bLegacyReport) {
+                if (node.name == 'DupViol') {
+                    mapToAnalyzer(node, "com.parasoft.xtest.cpp.analyzer.static.dupcode", false);
```

Comment:
I think we can put these if - else if to mapToAnalyzer(xxx) method. Just pass node parameter.